src/main/java/v/hryhoryk/onlinebookstore/controller/BookController.java

```diff
@@ -18,42 +23,92 @@
+            description = "Returns a book as per the ID")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200",
+                    description = "Successfully retrieved"),
+            @ApiResponse(responseCode = "404",
+                    description = "Not found - book by this ID was not found")
+    })
     @ResponseStatus(HttpStatus.OK)
     @GetMapping("/{id}")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getById(id);
     }
 
+    @Operation(summary = "Search books by parameters",
+            description = "Returns books by certain searching parameters")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200",
+                    description = "Successfully retrieved"),
+            @ApiResponse(responseCode = "404",
+                    description = "Not found - books by this searching parameters were not found")
+    })
     @ResponseStatus(HttpStatus.OK)
     @GetMapping("/search")
-    public List<BookDto> search(BookSearchParameters searchParameters) {
-        return bookService.search(searchParameters);
+    public List<BookDto> search(BookSearchParameters searchParameters, Pageable pageable) {
+        return bookService.search(searchParameters, pageable);
     }
 
+    @Operation(summary = "Create a new book",
+            description = "Adds a new book to the catalog")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201",
+                    description = "Successfully created"),
+            @ApiResponse(responseCode = "400",
+                    description = "Bad request - validation failed")
+    })
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.createBook(requestDto);
     }
 
+    @Operation(summary = "Update book by ID",
+            description = "Updates a book with the provided ID")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200",
+                    description = "Successfully updated"),
+            @ApiResponse(responseCode = "400",
+                    description = "Bad request - request data validation failed")
+    })
     @ResponseStatus(HttpStatus.OK)
     @PutMapping("/{id}")
     public BookDto updateById(@PathVariable Long id, @RequestBody CreateBookRequestDto book) {
         return bookService.updateById(id, book);
     }
 
+    @Operation(summary = "Delete book by ID",
+            description = "Deletes a book with the provided ID")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204",
+                    description = "Successfully deleted"),
+            @ApiResponse(responseCode = "404",
+                    description = "Not found - book by this ID was not found")
+    })
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
     public void deleteById(@PathVariable Long id) {
```

Comment:
you may also validate `@PathVariable` like:
```
@PathVariable @Positive Long id
```

in order to do so - annotate your controller with `@Validated` annotation